src/serlio/modifiers/PRTModifierAction.cpp

```diff
@@ -600,6 +601,89 @@ MStatus PRTModifierAction::updateUI(const MObject& node) {
 	return MStatus::kSuccess;
 }
 
+MStatus PRTModifierAction::updateDynamicEnums() {
+	for (auto& e : mEnums) {
+		if (e.mValuesAttr.length() > 0) {
+
+			const MString fullAttrName = e.mAttr.name();
+			const RuleAttribute ruleAttr = reverseLookupAttribute(fullAttrName.asWChar(), mRuleAttributes);
+
+			const std::wstring attrStyle = prtu::getStyle(ruleAttr.fqName).c_str();
+			std::wstring attrImport = prtu::getImport(ruleAttr.fqName).c_str();
+			if (!attrImport.empty())
+				attrImport += prtu::IMPORT_DELIMITER;
+
+			const std::wstring prefix = attrStyle + prtu::STYLE_DELIMITER + attrImport;
+
+			const wchar_t* valuesAttr = (MString(prefix.c_str()) + e.mValuesAttr).asWChar();
+			prt::Attributable::PrimitiveType type = mGenerateAttrs->getType(valuesAttr);
+
+			switch (type) {
+				case prt::Attributable::PT_STRING_ARRAY: {
+					size_t arr_length = 0;
+					const wchar_t* const* stringArray = mGenerateAttrs->getStringArray(valuesAttr, &arr_length);
+
+					for (short enumIndex = 0; enumIndex < arr_length; enumIndex++) {
+						std::wstring currString = stringArray[enumIndex];
+
+						// remove newlines from strings, because they break the maya UI
+						currString.erase(std::remove(currString.begin(), currString.end(), '\n'), currString.end());

```

Comment:
Maybe it would be better to escape them to `<nl>` or something. A newline in a enum name IMHO points to a buggy RPK, so if it is visible in the Maya UI it could help the user debugging this.

Reply:
ignore above comment but make sure this also trims \r in case the source file contains windows line endings

Reply:
Also trimming \r now.